provider/digest_provider.py

```
@@ -182,3 +183,73 @@ def put_digest(digest_id, data, settings, auth=True):
     url = settings.digest_endpoint.replace('{digest_id}', str(digest_id))
     return digest_put_request(url, settings.verify_ssl, digest_id, data,
                               digest_auth_key(settings, auth))
+
+
+def put_digest_to_endpoint(logger, digest_id, digest_content, settings):
+    "handle issuing the PUT to the digest endpoint"
+    put_status = None
+    try:
+        put_digest(digest_id, digest_content, settings)
+        put_status = True
+    except Exception as exception:
+        logger.exception(
+            "Exception issuing PUT to the digest endpoint for digest_id %s. Details: %s" %
+            (str(digest_id), str(exception)))
+    return put_status
```

We are covered in `except` blocks in the activities, so I'd say we can re-`raise` the exception rather than returning a boolean that could be go unchecked:

- happy path `None` is returned
- any error, an `Exception` is raised

---

A good observation here, thanks. 

I want to return something. 

Most of our requests are a `get`. There are a couple `post` in our existing providers, one being in the Fastly provider, where the return value if an exception is not raised is to return the `response`. 

I am also going to return the `response` if a digest `put` is successful, and this does away with the `put_status`. These affect both the `PublishDigest` and `IngestDigestToEndpoint` activities and their tests.

---

I agree if the response is interesting, it should be returned in the positive case.